uniswap/model/parts/uniswap_model.py

```diff
@@ -18,14 +18,39 @@ def p_actionDecoder(_params, substep, sH, s):
         'UNI_burn': 0,        
     }
     
+    # Token prices
+    pToken = s['DAI_balance'][data_counter] / s['ETH_balance'][data_counter]
+    yToken = uniswap_events['token_balance'][data_counter+1] / uniswap_events['eth_balance'][data_counter+1]
+    pETH = s['ETH_balance'][data_counter] / s['DAI_balance'][data_counter]
+    yETH = uniswap_events['eth_balance'][data_counter+1] / uniswap_events['token_balance'][data_counter+1]
+
+    eth_delta = uniswap_events['eth_delta'][data_counter]
+    token_delta = uniswap_events['eth_delta'][data_counter]
     event = uniswap_events['event'][data_counter]
-    
     action['action_id'] = event
     
     if event == 'TokenPurchase':
-        action['eth_sold'] = uniswap_events['eth_delta'][data_counter]
+        if classifier(eth_delta, _params['c_rule']) == 'Conv':
+            if eth_delta * pToken >= token_delta * (1 - _params['conv_tolerance']):
+                action['eth_sold'] = uniswap_events['eth_delta'][data_counter]
+            else:
+                pass
+        else:
+            if yToken > pToken:
+                action['eth_sold'] = uniswap_events['eth_delta'][data_counter]
+            else:
+                pass
```

Comment:
You could create a function to do this because below you almost have the same piece of code.

Reply:
Probably will do it, but actually the maths are not correct, so I need to fix it before.